src/cmake/modules/FindJXL.cmake

```diff
@@ -0,0 +1,27 @@
+# Find libjxl
+# Will define:
+# - JXL_FOUND
+# - JXL_INCLUDE_DIRS directory to include for libjxl headers
+# - JXL_LIBRARIES libraries to link to
+
+include (FindPackageHandleStandardArgs)
+
+find_path(JXL_INCLUDE_DIR
+  NAMES jxl/decode.h jxl/encode.h)
+mark_as_advanced(JXL_INCLUDE_DIR)
+
+find_library(JXL_LIBRARY
+  NAMES jxl)
+mark_as_advanced(JXL_LIBRARY)
+
+find_library(JXL_THREADS_LIBRARY
+  NAMES jxl_threads)
+mark_as_advanced(JXL_THREADS_LIBRARY)
+
+find_package_handle_standard_args(JXL
+  REQUIRED_VARS JXL_LIBRARY JXL_THREADS_LIBRARY JXL_INCLUDE_DIR)
+
+if(JXL_FOUND)
+  set(JXL_LIBRARIES ${JXL_LIBRARY} ${JXL_THREADS_LIBRARY})
+  set(JXL_INCLUDE_DIRS ${JXL_INCLUDE_DIR})
+endif(JXL_FOUND)
```

Comment:
If this is all found, in addition to those variables, please also set up targets in the modern CMake style. You can find an example of this [in our FindLibheif.cmake module](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/master/src/cmake/modules/FindLibheif.cmake#L43). 

Also, it is important for this module to set a JXL_VERSION so we can check version range. A good example of how to extract this from header files can be found [in our FindOpenVDB.cmake](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/master/src/cmake/modules/FindOpenVDB.cmake#L48)

Reply:
This comment still applies